src/autolua.zig

```diff
@@ -172,6 +172,12 @@ pub fn check(L: ?*lua.lua_State, idx: c_int, comptime T: type) T {
             // lua.lua_pop(L, 1);
             // const ptr = lua.lua_touserdata(L, idx);
         },
+        .Optional => |N| {
+            if (lua.lua_type(L, idx) < lua.LUA_TBOOLEAN) {
```

Comment:
does this work instead?:
```suggestion
            if (lua.lua_isnoneornil(L, idx)) {
```

Reply:
OH! I was searching for something like that for AGES but never found it! it seems like a rather simpler solution instead of my comparation with LUA_TBOOLEAN

And yes, yes it does